pkgs/development/libraries/libnbd/default.nix

```
@@ -32,8 +34,24 @@ stdenv.mkDerivation rec {
     libxml2
   ];
 
+  configureFlags = []
+    ++ lib.optionals withPython [ "--with-python-installdir=${placeholder "out"}/${python3.sitePackages}" ];
+
   installFlags = [ "bashcompdir=$(out)/share/bash-completion/completions" ];
 
+  postInstall = lib.optionalString withPython ''
+    mkdir -p ${placeholder "out"}/${python3.sitePackages}/nbd-${version}.dist-info
```

Instead of a heredoc, use a substituteAll template and something like

```suggestion
    install -Dm644 -t ${placeholder "out"}/${python3.sitePackages}/nbd-${version}.dist-info/METADATA ${./libndb-metadata}
```

---

That does seem more elegant.